dataactbroker/permissions.py

```diff
@@ -39,71 +49,111 @@ def inner(*args, **kwargs):
+    if not hasattr(g, 'user'):
+        return False
+    elif g.user.website_admin:
+        return True
+
+    # Ensure the permission exists and retrieve its ID and type
     try:
-        permission = PERMISSION_TYPE_DICT[perm]
+        permission_id = ALL_PERMISSION_TYPES_DICT[permission]
     except KeyError:
-        permission = 999
+        return False
+    permission_list = FABS_PERMISSION_ID_LIST if permission_id in FABS_PERMISSION_ID_LIST else DABS_PERMISSION_ID_LIST
 
+    # Loop through user's affiliations and return True if any match the permission
     for aff in g.user.affiliations:
-        # affiliation matches agency args
-        agency = (aff.cgac and aff.cgac.cgac_code == cgac_code) or (aff.frec and aff.frec.frec_code == frec_code)
-        # affiliation permissions are FABS
-        fabs = aff.permission_type_id == PERMISSION_SHORT_DICT['f']
-        # affiliation has permission higher than perm args
-        dabs_perms = aff.permission_type_id >= permission
-
-        if agency and ((perm == 'reader') or (dabs_perms and not fabs) or (perm == 'fabs' and fabs)):
-            matching_perms.append(aff)
+        # Check if affiliation agency matches agency args
+        if (aff.cgac and aff.cgac.cgac_code == cgac_code) or (aff.frec and aff.frec.frec_code == frec_code):
+            # Check if affiliation has higher permissions than permission args
+            aff_perm_id = aff.permission_type_id
+            if (permission == 'reader') or (aff_perm_id in permission_list and aff_perm_id >= permission_id):
+                return True
 
-    has_affil = hasattr(g, 'user') and len(matching_perms)
-    return admin or has_affil
+    return False
 
 
 def current_user_can_on_submission(perm, submission, check_owner=True):
-    """Submissions add another permission possibility: if a user created a submission, they can do anything to it,
-    regardless of submission agency"""
+    """ Submissions add another permission possibility: if a user created a submission, they can do anything to it,
+        regardless of submission agency
+
+        Args:
+            perm: string PermissionType value
+            submission: Submission object
+            check_owner: allows the functionality if the user is the owner of the Submission; default True
+
+        Returns:
+            Boolean result on whether the user has permissions greater than or equal to perm
+    """
     is_owner = hasattr(g, 'user') and submission.user_id == g.user.user_id
     return (is_owner and check_owner) or current_user_can(perm, submission.cgac_code, submission.frec_code)
 
 
-def requires_submission_perms(perm, check_owner=True):
-    """Decorator that checks the current user's permissions and validates that the submission exists. It expects a
-    submission_id parameter and will return a submission object"""
+def requires_submission_perms(perm, check_owner=True, check_fabs=False):
```

Comment:
If check_fabs is going to be a string in the end, maybe the default should be `''` or `None` just to keep it consistent.